src/source/Ice/ConnectionListener.c

```diff
@@ -292,9 +301,20 @@ PVOID connectionListenerReceiveDataRoutine(PVOID arg)
 
         // Need to unlock the mutex to ensure other racing threads unblock
         MUTEX_UNLOCK(pConnectionListener->lock);
-
-        // blocking call until resolves as a timeout, an error, a signal or data received
-        retval = POLL(rfds, nfds, CONNECTION_LISTENER_SOCKET_WAIT_FOR_DATA_TIMEOUT / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+        retval = 0;
+        if (nfds != 0) {
+            // TODO add support for socketpair() in windows
+            // This end of the socketpair has been added to the list of sockets polled
+            // in order to have a way to end the poll early from the destructor
+#ifndef _WIN32
+            rfds[nfds].fd = pConnectionListener->kickSocket[CONNECTION_LISTENER_KICK_SOCKET_LISTEN];
+            rfds[nfds].events = POLLIN;
+            rfds[nfds].revents = 0;
+            nfds++;
+#endif
```

Comment:
```
    struct pollfd rfds[CONNECTION_LISTENER_DEFAULT_MAX_LISTENING_CONNECTION];
```
The loop above iterates as follows:

```
for (i = 0, socketCount = 0; i < CONNECTION_LISTENER_DEFAULT_MAX_LISTENING_CONNECTION; i++)
```

Which means it is possible that `nfds` could end up equal to `CONNECTION_LISTENER_DEFAULT_MAX_LISTENING_CONNECTION` when the `for` loop exits, which means we would be trying to access out of bounds here.  

Perhaps a safe solution here is to define `rfds` as follows:

```
    struct pollfd rfds[CONNECTION_LISTENER_DEFAULT_MAX_LISTENING_CONNECTION + 1];
```